src/odemis/acq/fastem.py

```diff
@@ -962,49 +963,60 @@ def estimateTiledAcquisitionTime(stream, stage, area):
+            The x and y axes are aligned with the x and y axes of the ebeam scanner.
+        :param area: (float, float, float, float) xmin, ymin, xmax, ymax coordinates of the overview region.
+        :param live_stream: (StaticStream or None): StaticStream to be updated with each tile acquired,
+               to build up live the whole acquisition. NOT SUPPORTED YET.
+
+        :returns: (DataArray) The complete overview image.
+        """
+        fastem_conf.configure_scanner(stream.emitter, fastem_conf.OVERVIEW_MODE)
 
-    # The stage movement precision is quite good (just a few pixels). The stage's
-    # position reading is much better, and we can assume it's below a pixel.
-    # So as long as we are sure there is some overlap, the tiles will be positioned
-    # correctly and without gap.
-    overlap = STAGE_PRECISION / stream.emitter.horizontalFoV.value
-    logging.debug("Overlap is %s%%", overlap * 100)  # normally < 1%
+        # The stage movement precision is quite good (just a few pixels). The stage's
+        # position reading is much better, and we can assume it's below a pixel.
+        # So as long as we are sure there is some overlap, the tiles will be positioned
+        # correctly and without gap.
+        overlap = STAGE_PRECISION / stream.emitter.horizontalFoV.value
+        logging.debug("Overlap is %s%%", overlap * 100)  # normally < 1%
 
-    def _pass_future_progress(sub_f, start, end):
-        f.set_progress(start, end)
+        def _pass_future_progress(sub_f, start, end):
+            f.set_progress(start, end)
 
-    # Note, for debugging, it's possible to keep the intermediary tiles with log_path="./tile.ome.tiff"
-    sf = stitching.acquireTiledArea([stream], stage, area, overlap, registrar=REGISTER_IDENTITY,
-                                    focusing_method=FocusingMethod.NONE)
-    # Connect the progress of the underlying future to the main future
-    # FIXME removed to provide better progress update in GUI
-    #  When _tiledacq.py has proper time update implemented, add this line here again.
-    # sf.add_update_callback(_pass_future_progress)
-    das = sf.result()
+        # Note, for debugging, it's possible to keep the intermediary tiles with log_path="./tile.ome.tiff"
+        self.sf = stitching.acquireTiledArea([stream], stage, area, overlap, registrar=REGISTER_IDENTITY,
+                                        focusing_method=FocusingMethod.NONE)
+        # Connect the progress of the underlying future to the main future
+        # FIXME removed to provide better progress update in GUI
+        #  When _tiledacq.py has proper time update implemented, add this line here again.
+        # sf.add_update_callback(_pass_future_progress)
+        # Go one level deeper to cancel the future of the underlying acquisition
+        # TiledAcquisitionTask._cancelAcquisition will hence be called
+        f.task_canceller = self._cancel_acquisition
```

Comment:
Let's put this at the init. It will be clearer that placing it in the middle of the code.